logger/logger.go

```diff
@@ -0,0 +1,28 @@
+package logger
+
+import (
+	"go.uber.org/zap"
+)
+
+// Logger is a wrapped logger used by the stanza agent.
+type Logger struct {
+	*zap.SugaredLogger
+	emitter *Emitter
+}
+
+// AddReceiver will add a receiver to the logger.
+func (l *Logger) AddReceiver(receiver Receiver) {
+	l.emitter.AddReceiver(receiver)
+}
+
+// New will create a new logger.
+func New(sugared *zap.SugaredLogger) *Logger {
+	emitter := &Emitter{}
+	hooks := zap.Hooks(emitter.Emit)
```

Comment:
Per the zap docs, hooks should only be used for simple triggers that don't access the entry's internal fields. It recommends using a core instead (see [example](https://godoc.org/go.uber.org/zap#ex-WrapCore--Wrap)). Have you looked at implementing a core at all?

Reply:
Seems like if we make the Emitter into a core, we might be able to get away without changing the interface as well